From 53c41b7d2b6e2a37065e7963fa9c0938b9d936d2 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Wed, 3 Aug 2016 13:19:05 +0200
Subject: [PATCH] man: ip-link: Drop fou and gue related documentation

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1013584
Upstream Status: RHEL only

This is a partial revert of commit 90f1df715eb38 ("iproute: Descriptions
of fou and gue options in ip-link man pages") which, besides useful
additions, also documents options not supported in iproute as shipped
with RHEL7.
---
 man/man8/ip-link.8.in | 44 --------------------------------------------
 1 file changed, 44 deletions(-)

diff --git a/man/man8/ip-link.8.in b/man/man8/ip-link.8.in
index dc9728c..b9d9933 100644
--- a/man/man8/ip-link.8.in
+++ b/man/man8/ip-link.8.in
@@ -511,17 +511,6 @@ the following additional arguments are supported:
 .BI "ip link add " DEVICE
 .BR type " { " gre " | " ipip " | " sit " }"
 .BI " remote " ADDR " local " ADDR
-[
-.BR encap " { " fou " | " gue " | " none " }"
-] [
-.BR encap-sport " { " \fIPORT " | " auto " }"
-] [
-.BI "encap-dport " PORT
-] [
-.RB [ no ] encap-csum
-] [
-.RB [ no ] encap-remcsum
-]
 
 .in +8
 .sp
@@ -533,30 +522,6 @@ the following additional arguments are supported:
 - specifies the fixed local address for tunneled packets.
 It must be an address on another interface on this host.
 
-.sp
-.BR encap " { " fou " | " gue " | " none " }"
-- specifies type of secondary UDP encapsulation. "fou" indicates
-Foo-Over-UDP, "gue" indicates Generic UDP Encapsulation.
-
-.sp
-.BR encap-sport " { " \fIPORT " | " auto " }"
-- specifies the source port in UDP encapsulation.
-.IR PORT
-indicates the port by number, "auto"
-indicates that the port number should be chosen automatically
-(the kernel picks a flow based on the flow hash of the
-encapsulated packet).
-
-.sp
-.RB [ no ] encap-csum
-- specifies if UDP checksums are enabled in the secondary
-encapsulation.
-
-.sp
-.RB [ no ] encap-remcsum
-- specifies if Remote Checksum Offload is enabled. This is only
-applicable for Generic UDP Encapsulation.
-
 .in -8
 
 .TP
@@ -1293,15 +1258,6 @@ ip link help gre
 .RS 4
 Display help for the gre link type.
 .RE
-.PP
-ip link add name tun1 type ipip remote 192.168.1.1
-local 192.168.1.2 ttl 225 encap gue encap-sport auto
-encap-dport 5555 encap-csum encap-remcsum
-.RS 4
-Creates an IPIP that is encapsulated with Generic UDP Encapsulation,
-and the outer UDP checksum and remote checksum offload are enabled.
-
-.RE
 
 .SH SEE ALSO
 .br
-- 
1.8.3.1

